In [23]:
import pandas as pd
df = pd.read_table('virushostdb.daily.tsv')
phage_list=[]
for line in df['refseq id']:
    if ',' in line:
        line.split(',')
        accs=[acc.strip() for acc in line.split(',')]
        phage_list.extend(accs)
    else:
        phage_list.append(line.strip())
set1=set(phage_list)
print(len(set1))


17298


In [24]:
edwards = []
with open('explore/1-edwards/edwards2016.txt') as f:
    for line in f:
        phage_acc = line.strip().split('\t')[0]
        edwards.append(phage_acc)
edwards = set(edwards)
print(len(edwards))
edwards_uniq = []
for i in edwards:
    if i not in set1:
        edwards_uniq.append(i)
print(len(edwards_uniq))


820
12


In [25]:
phiads = []
with open('explore/2-PHIAF/data_pos.txt') as f:
    for line in f:
        phage_acc = line.strip().split(',')[0]
        phiads.append(phage_acc)
phiads = set(phiads)
print(len(phiads))
phiads_uniq = []
for i in phiads:
    if i not in set1:
        phiads_uniq.append(i)
print(len(phiads_uniq))


304
23


In [26]:
wangs = []
with open('explore/3-wang/wang_train_pairs.csv') as f:
    f.readline()
    for line in f:
        phage_acc = line.strip().split(',')[0]
        wangs.append(phage_acc)
wangs = set(wangs)
print(len(wangs))
wangs_uniq = []
for i in wangs:
    if i not in set1:
        wangs_uniq.append(i)
print(len(wangs_uniq))



826
3


In [34]:
all=[]
all.extend(wangs)
all.extend(phiads)
all.extend(edwards)
print(len(set(all)))

1771


In [30]:
yilou = []
yilou.extend(wangs_uniq)
yilou.extend(phiads_uniq)
yilou.extend(edwards_uniq)
yilou=set(yilou)    
print(len(yilou))
f = open("其他数据库有而VHDB没有的phage id.txt", "w")
f.write('\n'.join(sorted(yilou)))
f.close()

38


In [28]:
print(yilou)

{'JN116822', 'NC_005809', 'MN586033', 'NC_010763', 'NC_023591', 'JX901189', 'NC_021859', 'KR053194', 'MT657340', 'HG793132', 'MK061416', 'NC_042059', 'NC_003525', 'MH825699', 'NC_021778', 'MK814756', 'MG925343', 'KJ567041', 'NC_008355', 'KR337649', 'MN586063', 'MN234189', 'MN284906', 'MN813682', 'NC_005808', 'NC_010762', 'NC_023596', 'NC_007457', 'MW291017', 'NC_019781', 'KR053196', 'KX507345', 'MN234213', 'MK620896', 'MK433259', 'JQ896627', 'MN224566', 'MK977706'}


edwards,PHIAF,wang数据库共有1771个phage id
发现这些id在VHDB没有的phage id 有38个
genbank id：有14个VHDB没有的phage，其他的11个是VHDB已经有nc id了
NC id：13个都是被remove的过期id，VHDB实际有新的id。